### PR TITLE
Refactoring walking entity

### LINKED ISSUE #94: Refactoring walking entity
## This task is a part of the issue:
 - #63 

## Description
It needs to refactor the structure of "walking" based on the design discussion held by our team.

### Dev requirements and restrictions:
```swift
struct Walking {
    /// The unique identifier for the walking instance.
    let id: UUID

    /// The owner of the pet being walked.
    var owner: Owner?

    /// The pet sitter responsible for the walking service.
    let sitter: Sitter

    /// The current status of the walking service.
    let status: WalkingStatus

    /// The start time of the walking service.
    let start: Date

    /// The end time of the walking service.
    let end: Date

    /// The rating is given to owner by the pet sitter.
    let ownerRating: Int?

    /// The rating is given to sitter by the pet owner.
    let sitterRating: Int?
    
    /// The review given by the pet owner.
    let ownerReview: String?

    /// The review given by the pet sitter.
    let sitterReview: String?

    /// The owner additional notes to the sitter about any details of the walking service.
    var notes: String?

    /// The price for the walking service.
    let price: Double { get set }
}

enum WalkingStatus: String, Codable, CaseIterable {
    /// The walking is available for booking.
    case available = "Available"

    /// The walking is pending approval from the pet sitter.
    case pending = "Pending"

    /// The walking has been booked by owner.
    case booked = "Booked"

    /// The walking has been completed.
    case finished = "Done"
}
```

### PR DESCRIPTION
**Pull request by iBYarchitect**
## Summary
This PR is a part of the work to resolve #94 

## Changes
- Deleted supporting struct `Location` `Reviews` `Rating` and their tests for the `Walking` entity.
- Changed `Walking` and their tests.
- Changed `WalkingDummy` file.

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/Model/Walking/Walking.swift [hunk 1]
```diff
@@ -22,18 +22,21 @@ struct Walking: Codable, Identifiable {
...
+    let ownerRating: Int?
 
-    /// The owner and sitter reviews for the walking service.
-    let reviews: Review?
+    /// The rating is given to sitter by the pet owner.
+    let sitterRating: Int?
 
-    /// The ratings given by both the pet owner and the pet sitter
-    let rating: Rating?
+    /// The review given by the pet owner.
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Model/Walking/Walking.swift:31]:
```suggestion
    /// The review is given by the pet owner.
```

File: Woof/Woof/Model/Walking/Walking.swift [hunk 2]
```diff
@@ -22,18 +22,21 @@ struct Walking: Codable, Identifiable {
...
+    let ownerReview: String?
+
+    /// The review given by the pet sitter.
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Model/Walking/Walking.swift:34]:
```suggestion
    /// The review is given by the pet sitter.
```

Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Walking.swift]: fixed with 6602c66

File: Woof/Woof/Model/Walking/Walking.swift (hunk 1 above)
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Walking.swift]: fixed with 6602c66

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 ### Deleted
 
 - Delete `expired` status from enum `WalkingStatus` - [#91](https://github.com/IronFoundation/iOS-Woof/pull/91)
+- Deleted supporting struct `Location` `Reviews` `Rating` and their tests for the `Walking` entity  - [#95](https://github.com/IronFoundation/iOS-Woof/pull/95)
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply CHANGELOG.md]: fixed with 6602c66

File: Woof/Woof/Model/Walking/Walking.swift [hunk 4]
```diff
@@ -3,7 +3,7 @@ import Foundation
 /**
  Pet walking service provided by a pet sitter for pet owners.
  */
-struct Walking: Codable, Identifiable {
+struct Walking {
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/Model/Walking/Walking.swift]: I'm sorry, accidentally deleted this.
Returned with 6602c66

**Review by NadzeyaShpakouskaya — APPROVED** [review]
LGTM

**Pull request merged**